src/k8s/pkg/k8sd/app/hooks_remove.go

```diff
@@ -92,13 +91,9 @@ func (a *App) onPreRemove(ctx context.Context, s state.State, force bool) (rerr
 				log.Error(err, "Failed to create k8s-dqlite client: %w")
 			}
 
-			log.Info("Cleaning up k8s-dqlite directory")
-			if err := os.RemoveAll(snap.K8sDqliteStateDir()); err != nil {
-				return fmt.Errorf("failed to cleanup k8s-dqlite state directory: %w", err)
-			}
 		case "external":
 			log.Info("Cleaning up external datastore certificates")
-			if _, err := setup.EnsureExtDatastorePKI(snap, &pki.ExternalDatastorePKI{}); err != nil {
+			if err := setup.RemoveExtDatastorePKIFiles(snap); err != nil {
```

Comment:
This is not required. The files are deleted if the respective PKI file is not set.
See https://github.com/canonical/k8s-snap/blob/28919a246052d403b865cada03b25f35cc3f6a74/src/k8s/pkg/k8sd/setup/certificates.go#L15

Reply:
Ack. The name `EnsureExtDatastorePKI` is a bit confusing, the first thing that would come to mind is "ensure the files exists". Reverted this change.